### PR TITLE
[1.19.3] Add helper methods to access the set of loaded sprite locations

### PR DESCRIPTION
**Pull request by Minecraftschurli**
This PR adds a readonly way to lookup the resource locations of all sprites in a texture atlas

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/resources/TextureAtlasHolder.java.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    public void close() {
+       this.f_118884_.m_118329_();
+    }
++
++   /**
++    * {@return the set of sprites in the atlas.}
++    */
++   public java.util.Set<ResourceLocation> getTextureLocations() {
++      return f_118884_.getTextureLocations();
++   }
```
Comment (sciwhiz12) [inline patches/minecraft/net/minecraft/client/resources/TextureAtlasHolder.java.patch]: Why not return the texture atlas instead, and let the user call the other newly-added helper method to access the sprites?

**Review by sciwhiz12 — COMMENTED** [review]
Could you please provide use cases for these helper methods?

File: src/main/resources/META-INF/accesstransformer.cfg [hunk 2]
```diff
@@ -135,6 +135,7 @@ public net.minecraft.client.renderer.texture.SpriteContents f_243731_ # byMipLev
 default net.minecraft.client.renderer.texture.SpriteContents f_244575_ # animatedTexture
 default net.minecraft.client.renderer.texture.SpriteContents m_245088_()I # getFrameCount
 public net.minecraft.client.resources.ClientPackSource m_246691_(Ljava/nio/file/Path;)Lnet/minecraft/server/packs/VanillaPackResources; # createVanillaPackSource
+protected net.minecraft.client.resources.TextureAtlasHolder f_118884_ # textureAtlas
```
Comment (sciwhiz12) [inline src/main/resources/META-INF/accesstransformer.cfg:138]: Why `protected` and not `public`? _What is the usecase?_

**Review by sciwhiz12 — COMMENTED** [review]
Again, please **provide use cases** for the helper method and the access to the field.

File: src/main/resources/META-INF/accesstransformer.cfg (hunk 2 above)
Comment (Minecraftschurli) [reply src/main/resources/META-INF/accesstransformer.cfg:138]: my usecase is custom atlases that previously had to provide the list of resource locations to load and now have no idea what is in the atlas after loading (which is data I need)
the only vanilla atlases that use this MobEffectTextureManager and PaintingTextureManager
if you think public would be better then I'll change that

**Pull request merged**